pg_search/src/postgres/customscan/aggregatescan/scan_state.rs

```diff
@@ -189,66 +182,80 @@ impl AggregateScanState {
                 aggregate_values: row,
             }];
         }
-        // GROUP BY - extract bucket results
+        // GROUP BY - extract nested bucket results recursively
         let mut rows = Vec::new();
 
-        // Navigate to the bucket results
-        let bucket_name = "group_0"; // For now, we only support single grouping column
-        let bucket_results = result
-            .get(bucket_name)
+        self.extract_bucket_results(&result, 0, &mut Vec::new(), &mut rows);
+
+        // Sort according to ORDER BY
+        self.sort_rows(&mut rows);
+        rows
+    }
+
+    #[allow(unreachable_patterns)]
```

Comment:
Rather than doing this, would suggest explicitly matching the relevant patterns and panicing. I think that it makes it more obvious that there are potential error cases. And if the case is supposed to "definitely be impossible", then can do:
```rust
  x => unreachable!("{x:?} already handled.");
```